sound/soc/sof/imx/imx8.c

```diff
@@ -425,6 +508,14 @@ static struct snd_soc_dai_driver imx8_dai[] = {
 },
 };
 
+static int imx8_dsp_set_power_state(struct snd_sof_dev *sdev,
+				    const struct sof_dsp_power_state *target_state)
+{
+	sdev->dsp_power_state = *target_state;
```

Comment:
Are all power transitions allowed or valid?

Reply:
This is taken care of by the SOF core.